src/main/java/bio/terra/service/dataset/flight/ingest/IngestValidateGcpRefsStep.java

```diff
@@ -41,8 +41,10 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     for (Column column : table.getColumns()) {
       if (column.isFileOrDirRef()) {
         List<String> refIdArray = bigQueryDatasetPdao.getRefIds(dataset, stagingTableName, column);
-        List<String> badRefIds = fileDao.validateRefIds(dataset, refIdArray);
-        badRefIds.forEach(id -> invalidRefIds.add(new InvalidRefId(id, column.getName())));
+        if (!refIdArray.isEmpty()) {
```

Comment:
I think this check should be done inside `FireStoreDao.validateRefIds()` instead of in the caller, since it's an optimization that method can do regardless of who calls `validateRefIds()`. It could even be pushed one layer down into `FireStoreDirectoryDao.validateRefIds()`.